File: src/main/java/com/project/QuanLyMyCay/entity/Product.java
package com.project.QuanLyMyCay.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "name", nullable = false, length = 255) // VARCHAR(255), NOT NULL
    private String name;

    @Column(name = "price", nullable = false) // NOT NULL
    private double price;

    @Column(name = "description", length = 65535) // LONGTEXT
    private String description;

    @Column(name = "thumbnail", length = 255) // VARCHAR(255)
    private String thumbnail;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false) // NOT NULL foreign key
    private Category category;

    @Column(name = "status", nullable = false, columnDefinition = "TINYINT DEFAULT 1") // Default = 1
    private boolean status;
}

File: FE/lib/model/Product.dart
import 'Category.dart';

class Product {
  final int id;
  final String name;
  final double price;
  final String description;
  final String? thumbnail; // Có thể không có thumbnail
  final Category category; // Loại category, cần thêm model Category
  final bool status;

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    this.thumbnail,
    required this.category,
    required this.status,
  });

  // Hàm tạo đối tượng Product từ JSON
  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      price: json['price'],
      description: json['description'],
      thumbnail: json['thumbnail'],
      category: Category.fromJson(json['category']), // Cần tạo category từ JSON
      status: json['status'],
    );
  }
}
